src/libraries/NetCoreAppLibrary.props

```diff
@@ -146,6 +146,7 @@
       System.Xml.XPath;
       System.Xml.XPath.XDocument;
     </NetCoreAppLibrary>
+    <NetCoreAppLibrary Condition="'$(ClrNativeAOT)' == 'true'">$(NetCoreAppLibrary)System.Private.DeveloperExperience.Console;</NetCoreAppLibrary>
```

Comment:
Is this going to do the right thing when all runtime flavors are building as part of a single build?

Reply:
Flavors, yes, but variants, probably not.

I don't think we can build CoreCLR Native AOT and CoreCLR JIT-based at the same time, given libraries are going to be structured differently between those two (the S.D.StackTrace difference discussed in the other pull request comes to mind as another example where we just need to produce a different assembly).

S.P.DeveloperExperience.Console can only be built against the CoreLib that has the DeveloperExperience contract.

Reply:
I think we need to get to a place where simple `build.cmd` invocation can build everything - CoreCLR-JIT, CoreCLR-NativeAOT, Mono, .... . 

If it is difficult to do that immediately, I am fine with with temporary workarounds,

Reply:
We could redo the DeveloperExperince class to be just an `Action<string>` delegate. Then we don't require anything special in the CoreLib and can build this library against any CoreLib.

Reply:
Seems like this is problematic for the All_Configurations build anyway (whatever that is).

Reply:
@jkotas  I was thinking a bit about this. The options I see:

1. Add the a DeveloperExperience contract to the shared corelib so that Mono and CoreCLR also have it and we can build the assembly against their corelibs too. Seems a bit dirty.
2. Get rid of the DeveloperExperience assembly and just p/invoke into the OS from CoreLib. Seems rather annoying because suddenly we'll need to deal with "how do I write unicode characters into stderr?". Last time I had to do that, it was surprisingly complicated and involved converting codepages (at least on Windows).
3. Just `AppContext.SetData("System.Runtime.ExceptionServices.WriteStackTraceString", new Action<string>(Console.Error.WriteLine));` in the DeveloperExperience LibraryInitializer and grab that thing out of AppContext in CoreLib.

Number 3 seems the most preferable to me.

Reply:
I agree. 3 sounds good.